packages/mobile/src/verify/VerificationLoadingScreen.tsx

```diff
@@ -5,77 +5,83 @@ import colors from '@celo/react-components/styles/colors'
 import CancelButton from 'src/components/CancelButton'
 import Carousel, { CarouselItem } from 'src/components/Carousel'
 import DevSkipButton from 'src/components/DevSkipButton'
 import { Namespaces } from 'src/i18n'
 import LoadingSpinner from 'src/icons/LoadingSpinner'
-import {
-  cancelVerification,
-  receiveAttestationMessage,
-  startVerification,
-} from 'src/identity/actions'
-import { AttestationCode } from 'src/identity/verification'
+import { cancelVerification, startVerification } from 'src/identity/actions'
+import { VerificationStatus } from 'src/identity/verification'
+import { navigate, navigateBack } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { RootState } from 'src/redux/reducers'
-import { currentAccountSelector } from 'src/web3/selectors'
+import Logger from 'src/utils/Logger'
+import VerificationFailedModal from 'src/verify/VerificationFailedModal'
+
+const TAG = 'VerificationLoadingScreen'
 
 interface StateProps {
-  numberVerified: boolean
   e164Number: string
-  account: string | null
-  attestationCodes: AttestationCode[]
-  numCompleteAttestations: number
-  verificationFailed: boolean
-  underlyingError: ErrorMessages | null | undefined
+  verificationStatus: VerificationStatus
 }
 
 interface DispatchProps {
   startVerification: typeof startVerification
   cancelVerification: typeof cancelVerification
-  receiveVerificationMessage: typeof receiveAttestationMessage
-  showError: typeof showError
-  hideAlert: typeof hideAlert
 }
 
 type Props = StateProps & DispatchProps & WithNamespaces
 
 const mapDispatchToProps = {
   startVerification,
   cancelVerification,
-  receiveVerificationMessage: receiveAttestationMessage,
-  showError,
-  hideAlert,
 }
 
 const mapStateToProps = (state: RootState): StateProps => {
   return {
-    numberVerified: state.app.numberVerified,
     e164Number: state.account.e164PhoneNumber,
-    attestationCodes: state.identity.attestationCodes,
-    numCompleteAttestations: state.identity.numCompleteAttestations,
-    verificationFailed: state.identity.verificationFailed,
-    account: currentAccountSelector(state),
-    underlyingError: errorSelector(state),
+    verificationStatus: state.identity.verificationStatus,
   }
 }
 
 class VerificationLoadingScreen extends React.Component<Props> {
   static navigationOptions = { header: null }
 
+  componentDidMount() {
+    BackHandler.addEventListener('hardwareBackPress', this.handleBackButton)
+    this.props.startVerification()
+  }
+
+  componentDidUpdate() {
+    if (this.props.verificationStatus === VerificationStatus.Done) {
+      navigate(Screens.VerificationSuccessScreen)
+    } else if (this.props.verificationStatus === VerificationStatus.RevealingNumber) {
+      navigate(Screens.VerificationInterstitialScreen)
+    }
+  }
+
+  componentWillUnmount() {
+    BackHandler.removeEventListener('hardwareBackPress', this.handleBackButton)
```

Comment:
So I'm not sure here exactly how this works, but I don't know if unmount is called if you push another screen onto the stack, so if that happens we could have multiple listeners at the same time. My intuition is that the latest one would take precedence, but just wanted to flag

Reply:
Good catch. I'll look into it and see if there's a better solution